src/nix/flake.cc

```diff
@@ -272,25 +272,40 @@ struct CmdFlakeCheck : FlakeCommand
         auto state = getEvalState();
         auto flake = lockFlake();
 
+        bool hasErrors = false;
+        auto throw_ = [&](const Error & e) {
```

Comment:
Maybe call this `reportError`, "throw" suggests that it won't return.

Reply:
Oh yes, you’re right. Fixed in https://github.com/NixOS/nix/pull/4878/commits/7d651f5c3fc721b8957a2266e5e14e99bdb77077